### PR TITLE
Adding basic example Flexvolume drivers and DaemonSet deployment example.

### LINKED ISSUE #59148: Add Flexvolume DaemonSet deployment examples to the examples directory
> /kind feature

Currently incomplete examples of DaemonSet deployment is only linked in the Flexvolume readme. A more complete example requiring minimal user interaction should be included in the `example/volume/flexvolume` directory.

### PR DESCRIPTION
**Pull request by verult**
**What this PR does / why we need it**: More example Flexvolume drivers as reference for driver authors, and an example driver deployment workflow.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #59148

/sig storage
/assign @chakri-nelluri @zmerlynn 
/cc @wongma7 
/release-note-none

### CODE REVIEW

File: examples/volumes/flexvolume/deploy/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+
+# TODO change to your desired driver.
+VENDOR=k8s
```
Comment (chakri-nelluri) [inline examples/volumes/flexvolume/deploy/deploy.sh]: Allow users to set this as environment variable and fail if they are not set? That way users can reuse this example.

File: examples/volumes/flexvolume/deploy/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+# Assuming the single driver file is located at /$DRIVER inside the DaemonSet image.
+
+driver_dir=$VENDOR${VENDOR:+"~"}${DRIVER}
+if [ ! -d "/flexmnt/$driver_dir" ]; then
+  mkdir "/flexmnt/$driver_dir"
+fi
+
+cp "/$DRIVER" "/flexmnt/$driver_dir/.$DRIVER"
+mv -f "/flexmnt/$driver_dir/.$DRIVER" "/flexmnt/$driver_dir/$DRIVER"
```
Comment (chakri-nelluri) [inline examples/volumes/flexvolume/deploy/deploy.sh]: Use a tmp identifier so that users can realize they are tmp files Ex: .tmp_$DRIVER?

**Review by chakri-nelluri — COMMENTED** [review]
Minor comments.

**Comment by verult** [comment]
Addressed comments

File: examples/volumes/flexvolume/deploy/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+
+# TODO change to your desired driver.
+VENDOR=${VENDOR:-k8s}
+DRIVER=${DRIVER:-dummy}
```
Comment (chakri-nelluri) [inline examples/volumes/flexvolume/deploy/deploy.sh]: Hmm.. shall we fail if these variables are not set? How about we add a check and exit the container with a log? These variables have to be defined.

**Review by chakri-nelluri — COMMENTED** [review]
One minor comment. PTAL when you get a chance.

File: examples/volumes/flexvolume/deploy/deploy.sh (hunk 3 above)
Comment (verult) [reply examples/volumes/flexvolume/deploy/deploy.sh]: I'd like to make it minimal effort for users to get a basic example running. Currently the user only has to copy the driver into the appropriate directory, then build and deploy.

I'll add a comment about that in the README

Comment (chakri-nelluri) [reply examples/volumes/flexvolume/deploy/deploy.sh]: ACK

**Review by chakri-nelluri — COMMENTED** [review]
/LGTM

**Comment by verult** [comment]
/cc @brendandburns @eparis @thockin 
for approval of the `examples/` directory. Thanks!

**Pull request merged**